base/array.jl

```diff
@@ -791,9 +791,10 @@ function find(testf::Function, A::AbstractArray)
     I
 end
 
-function find(A::AbstractArray)
-    nnzA = countnz(A)
-    I = similar(A, Int, nnzA)
+function find(A)
+    cA = typeof(A) <: AbstractArray ? A : collect(A)
```

Comment:
I don't think calling `collect` that's a good idea. For example, `find(1:10000000)` is going to kill your computer, while the current code works perfectly fine. People will have to call `collect` manually if they want to work on iterables on which one can only pass over once.

So just remove the `::AbstractArray` assertion and replace `similar(Int, nnzA)` with `Vector{Int}(nnzA)`, and you should be fine.

(BTW: it's always better to avoid branches on types inside functions, and create a separate method which calls the more specific one after calling `collect`.)